drivers/include/sdcard_spi.h

```diff
@@ -0,0 +1,359 @@
+#define SD_CMD_59_IDX 59    /* Turns the CRC option on or off. A '1'in the CRC option bit will turn the option on, a '0'will turn it off */
+
+#define SD_CMD_8_VHS_2_7_V_TO_3_6_V 0b00000001
+#define SD_CMD_8_CHECK_PATTERN      0b10110101
+#define SD_CMD_ARG_NONE   0x00000000
+#define SD_ACMD_41_ARG_HC 0x40000000
+#define SD_CMD_59_ARG_ENABLE  0x00000001
+#define SD_CMD_59_ARG_DISABLE 0x00000000
+
+/* see sd spec. 7.3.3 Control Tokens */
+#define SD_DATA_TOKEN_CMD_17_18_24 0b11111110
+#define SD_DATA_TOKEN_CMD_25       0b11111100
+#define SD_DATA_TOKEN_CMD_25_STOP  0b11111101
+
+#define SD_SIZE_OF_CID_AND_CSD_REG 16
+#define SD_GET_CSD_STRUCTURE(CSD_RAW_DATA) ((CSD_RAW_DATA)[0] >> 6)
+#define SD_CSD_V1 0
+#define SD_CSD_V2 1
+#define SD_CSD_VUNSUPPORTED -1
+
+/* the retry counters below are used as timeouts for specific actions.
+   The values may need some adjustments to either give the card more time to respond
+   to commands or to achieve a lower delay / avoid infinite blocking. */
+#define R1_POLLING_RETRY_CNT    10000
+#define SD_DATA_TOKEN_RETRY_CNT 10000
+#define INIT_CMD_RETRY_CNT      1000
+#define INIT_CMD0_RETRY_CNT     3
+#define SD_WAIT_FOR_NOT_BUSY_CNT 10000   /* setting this to -1 leads to full blocking until the card isn't busy anymore */
+#define SD_BLOCK_READ_CMD_RETRIES 10    /* this only accounts for sending of the command not the whole read transaction! */
+#define SD_BLOCK_WRITE_CMD_RETRIES 10   /* this only accounts for sending of the command not the whole write transaction! */
+
+#define SD_HC_FIXED_BLOCK_SIZE 512
+#define SD_CSD_V2_C_SIZE_BLOCK_MULT 1024 /* memory capacity in bytes = (C_SIZE+1) * SD_CSD_V2_C_SIZE_BLOCK_MULT * BLOCK_LEN */
+
+
+#define SD_CARD_DEFAULT_SPI_CONF SPI_CONF_FIRST_RISING
+#define SD_CARD_SPI_SPEED_PREINIT SPI_SPEED_100KHZ  /* this speed setting is only used while the init procedure is performed */
+#define SD_CARD_SPI_SPEED_POSTINIT SPI_SPEED_10MHZ  /* after init procedure is finished the driver auto sets the card to this speed */
+
+#define SD_CARD_DUMMY_BYTE 0xFF
+
+#ifndef NUM_OF_SD_CARDS
+    #define NUM_OF_SD_CARDS 1
+#endif
+
+typedef enum {
+    SD_V2,
+    SD_V1,
+    MMC_V3,
+    SD_UNKNOWN
+} sd_version_t;
+
+typedef enum {
+    SD_INIT_START,
+    SD_INIT_SPI_POWER_SEQ,
+    SD_INIT_SEND_CMD0,
+    SD_INIT_SEND_CMD8,
+    SD_INIT_CARD_UNKNOWN,
+    SD_INIT_SEND_ACMD41_HCS,
+    SD_INIT_SEND_ACMD41,
+    SD_INIT_SEND_CMD1,
+    SD_INIT_SEND_CMD58,
+    SD_INIT_SEND_CMD16,
+    SD_INIT_ENABLE_CRC,
+    SD_INIT_READ_CID,
+    SD_INIT_READ_CSD,
+    SD_INIT_SET_MAX_SPI_SPEED,
+    SD_INIT_FINISH
+} sd_init_fsm_state_t;
+
+typedef enum {
+    SD_RW_OK = 0,
+    SD_RW_NO_TOKEN,
+    SD_RW_TIMEOUT,
+    SD_RW_RX_TX_ERROR,
+    SD_RW_WRITE_ERROR,
+    SD_RW_CRC_MISMATCH,
+    SD_RW_NOT_SUPPORTED
+} sd_rw_response_t;
+
+struct {
+    uint8_t MID;            /* Manufacturer ID */
+    char OID[2];            /* OEM/Application ID*/
+    char PNM[5];            /* Product name */
```

Comment:
Please adhere to [coding conventions][1]

[1]: https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions#absolute-values